src/WooCommerce.php

```diff
@@ -1513,4 +1514,21 @@ private static function getRepricingOptionsDropdownValues(\WC_Product $product)
     return $dropdown_values;
   }
 
+    /**
+   * Invalidate the object cache in Woocommerce order status manager
+   * 
+   * @param int $post_id
+   * 
+   * @param \WP_Post $post
+   *   all post types - we're checking for wc_order_status post type
+   * 
+   * @return void
+   * 
+   */
+  private static function save_post_custom_status_manager(int $post_id, \WP_Post $post) {
+    if($post->post_type === 'wc_order_status' && wp_cache_get('wc_order_status_manager_order_status_posts')) {
```

Comment:
I think we can use 
https://developer.wordpress.org/reference/hooks/save_post_post-post_type/
to target specifically that post type and that will make the if statement obsolete.